src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -658,13 +658,14 @@ public <T extends Serializable> T runRemotely(Step2<T> s) throws Throwable {
         HttpURLConnection conn = (HttpURLConnection) endpoint("step").openConnection();
         conn.setRequestProperty("Content-Type", "application/octet-stream");
         conn.setDoOutput(true);
-        Init2.writeSer(conn.getOutputStream(), Arrays.asList(token, s, getUrl()));
-        Throwable error;
-        T object;
+
+        Init2.writeSer(conn.getOutputStream(), new InputPayload(token, s, getUrl()));
         try {
-            List<Object> result = (List) Init2.readSer(conn.getInputStream(), null);
-            object = (T) result.get(0);
-            error = (Throwable) result.get(1);
+            OutputPayload result = (OutputPayload) Init2.readSer(conn.getInputStream(), null);
+            if (result.error != null) {
+                throw new StepException(result.error);
```

Comment:
Why not just throw `result.error` directly? Do we actually need `StepException` at all?